lightning/src/ln/peers/handshake/acts.rs

```diff
@@ -1,42 +1,171 @@
-pub const ACT_ONE_LENGTH: usize = 50;
-pub const ACT_TWO_LENGTH: usize = 50;
-pub const ACT_THREE_LENGTH: usize = 66;
-
-/// Wrapper for the first act message
-pub struct ActOne(
-	pub(super) [u8; ACT_ONE_LENGTH]
-);
+//! Helper library for working with NOISE handshake Act data. Contains utilities for passing Act
+//! objects around and building them from received data.
+//! Act objects are thin wrappers about raw arrays for stack-based processing as well as convenient
+//! coercion to slices for flexible use by the higher-level modules.
 
-/// Wrapper for the second act message
-pub struct ActTwo(
-	pub(super) [u8; ACT_TWO_LENGTH]
-);
+use std::{cmp, ops};
 
-/// Wrapper for the third act message
-pub struct ActThree(
-	pub(super) [u8; ACT_THREE_LENGTH]
-);
+pub const ACT_ONE_TWO_LENGTH: usize = 50;
+pub const ACT_THREE_LENGTH: usize = 66;
+pub const EMPTY_ACT_ONE: ActOne = [0; ACT_ONE_TWO_LENGTH];
```

Comment:
@arik-so Do you think we should fulfill our initial ciphertext buffers with randomness instead of zero'ing them ? In case of Chacha failure biasing towards zero, it may mask it.

Reply:
I think an issue with using randomness is that tests could become non-deterministic. We can, however, go the BOLT way and use a value other than 0 for the initial fill

Reply:
@arik-so Can you point me to an example in the BOLT that I can reference?

Reply:
see `session_key` or `associated_data` here: https://github.com/lightningnetwork/lightning-rfc/blob/master/04-onion-routing.md#returning-errors-1

Reply:
We can have a static (`#[cfg(test)]`) value for our testing framework ? Coverage should  be the same as we don't control flow on this value. Any production bug due to it should be caught equally by test framework if it's covered ?

Reply:
I'd like to propose pushing this to a security feature that is added as a separate PR. I've included it here and will move those to Github Issues for easier tracking. https://docs.google.com/document/d/1dTUac58ikbODDMffBSF1C7hwPNd_5t1qvzSHVXSxJCs/edit?usp=sharing

This seems like a good idea overall, but isn't a regression from master and needs a bit of design to see how we get the bytes down to this layer and test it properly.

In that story, all the tests will just push in known random bytes just like they do for the ephemeral key generation now. Shouldn't need any cfg(test) work.

 Does that seem like an OK plan to help move this along?

Reply:
Sure let's not withhold on it and think about it latter.

Reply:
Tracked https://github.com/rust-bitcoin/rust-lightning/issues/700